packages/playwright-core/src/remote/playwrightServer.ts

```diff
@@ -68,12 +68,20 @@ export class PlaywrightServer {
   }
 
   async listen(port: number = 0): Promise<string> {
-    const server = http.createServer((request, response) => {
+    let wsEndpoint: string | undefined = undefined;
+    const server = http.createServer((request: http.IncomingMessage, response: http.ServerResponse) => {
+      if (request.method === 'GET' && request.url === '/json') {
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({
+          wsEndpoint,
+        }));
+        return;
+      }
       response.end('Running');
     });
     server.on('error', error => debugLog(error));
 
-    const wsEndpoint = await new Promise<string>((resolve, reject) => {
+    wsEndpoint = await new Promise<string>((resolve, reject) => {
```

Comment:
IIUC, this will always be `localhost` or similar. However, when connecting to `<ip>:<port>` and fetching `<ip>:<port>/json`, we should preserve the ip/port. Perhaps just make `wsEndpoint` a relative path instead?